src/Api/JsonRpc/Client.php

```diff
@@ -374,7 +377,7 @@ public function addTestResult($name, $result, $containingClass, $text = "", $exe
             $testCase->setCaseStatusId($this->caseStatusId);
             $testCase->setIsAutomated(true);
             $testCase->setPriorityId($this->priorityId);
-            $testCase->setAuthorId(1);
+            $testCase->setAuthorId($this->user->getId());
```

Comment:
I looked over other plugins but couldnt found a use of "author" for the new test cases. Here $this->user is the user whose credentials are used for the calls to the API(defined in tcms.conf).

Reply:
Hmmm, IDK what's happening here. `$testCase->setAuthorId()` will change the value of an object attribute. However I don't understand how that gets propagated to the RPC call. Where do I look ? 

To answer a question you didn't ask, here's how creating a new TestCase is handled by the RPC layer:
https://github.com/kiwitcms/Kiwi/blob/master/tcms/rpc/api/testcase.py#L255

if a value for `author` is not provided explicitly it is taken from the HTTP request and defaults to the currently logged user, which is the same account sending the RPC request in the first place. 

From what I can tell you don't need the `setAuthorId` methods anywhere in the source code.

Reply:
The testCase object is used [here](https://github.com/kiwitcms/phpunit-plugin/blob/a638877921e6b97895e6e0b419e185c18cbbd64e/src/Api/JsonRpc/Client.php#L428). After that we call [testCaseRepository](https://github.com/kiwitcms/phpunit-plugin/blob/a638877921e6b97895e6e0b419e185c18cbbd64e/src/Api/JsonRpc/Client.php#L392) where we extract the data from the object and send to the KiwiTCMS [link](https://github.com/kiwitcms/phpunit-plugin/blob/a638877921e6b97895e6e0b419e185c18cbbd64e/src/Api/JsonRpc/Repository/TestCaseRepository.php#L69)

If the logged user is used I agree that we dont need this setter at all.